### PR TITLE
quick fixes to pettingzoo AgentID and pre-commit pylint

### PR DESCRIPTION
**Pull request by kywch**
1. bumped pettingzoo to 1.19.0, which doesn't mess with gym==0.23.0
2. moved gym to `install_requires` from `cleanrl` since nmmo core uses gym
3. pinned all package versions, except torch, tensorboard
4. during pre-commit checks, install all dependencies to make the checks work. (it seems that specifying `known-third-party=` in pylint.cfg doesn't fix this error)

### CODE REVIEW

**Comment by kywch** [comment]
Getting no more `Run failed: Pylint - v1.6 (673a284)` emails!

**Comment by kywch** [comment]
@daveey 's `git-pr.sh` always makes a PR to the carper branch, so I added an option not to when `n` is entered to the prompt: `Do you like to create a PR to the CarperAI/nmmo-environment repo? (y/n)`

It only checks `n`, so pressing <enter> will just create a PR as usual.

File: scripts/git-pr.sh [hunk 1]
```diff
@@ -66,9 +66,10 @@ echo "Pushing topic branch to origin..."
 git push -u origin $branch_name
 
 # Generate a Github pull request
-echo "Generating Github pull request..."
-pull_request_url="https://github.com/CarperAI/nmmo-environment/compare/$MASTER_BRANCH...CarperAI:nmmo-environment:$branch_name?expand=1"
-
-echo "Pull request URL: $pull_request_url"
-
+read -p "Do you like to create a PR to the CarperAI/nmmo-environment repo? (y/n) " ans
```
Comment (daveey) [inline scripts/git-pr.sh]: Do we need to ask the user if they want to create a PR? All this does is generate a URL so there is no reason to bother the user with this question.

If we do want to ask the user, let's do it before pushing the branch?

**Review by daveey — APPROVED** [review]

File: scripts/git-pr.sh (hunk 1 above)
Comment (kywch) [reply scripts/git-pr.sh]: Oh, I thought it was actually making a PR and commented out. If it's just providing the url, I agree that there is no reason to bother. 

Thanks, David!

**Pull request merged**